expan/core/experiment.py

```diff
@@ -287,11 +288,70 @@ def _quantile_filtering(self, data, kpis, percentile, threshold_type):
+        :param weights: list of dicts with weights/split percentage per each variant 
+                        ([{'value':<variant>,'weight':<weight>}, ...])
+        :param min_counts: minimum number of observed and expected frequencies (should be at least 5), see 
+                            http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        :param alpha: significance level, 0.05 by default
+        :return: True(if split is consistent with the given split) or
+                 False(if split is not consistent with the given split)
+        :rtype:  Boolean, float, float
+        """
+        if not variant_column:
+            logger.warning("Variant split check was cancelled since input variant column is empty or doesn't exist.")
+            return None
+        if not weights:
+            logger.warning("Variant split check was cancelled since input weights are empty or doesn't exist.")
+            return None
+        if len(weights) <= 1 or len(variant_column) <= 1:
+            logger.warning("Variant split check was cancelled since input weights or the number if categories "
+                           "is less than 2.")
+            return None
+
+        # Count number of observations per each variant
+        variant_column = pd.Series(variant_column).fillna(0)
+        variants_counts = variant_column.value_counts()
+        observed_counts = pd.DataFrame(variants_counts, columns=['observed']).fillna(0)
+        # Ensure at least a frequency of 5 at every location in observed_counts, otherwise drop the category
+        # see http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        observed_freqs = observed_counts[observed_counts >= min_counts].dropna(axis=0)
+
+        # If all categories were dropped or if there are less than 2 categories left we can't conduct the test.
+        if observed_freqs.empty or len(observed_freqs) < 2:
+            logger.warning("Category frequencies are empty or number of categories is less than 2. "
+                           "Chi-square test is not applicable.")
+            return None
+
+        observed_freqs.index.name = 'value'
+
+        # Calculate expected counts given corresponding weights
+        total_count = observed_freqs.sum().sum()
+        expected_freqs = pd.DataFrame(weights)
+
+        # Join all values in one data frame, so that each observed category count corresponds to the correct expected
+        # count defined by the corresponding weight or split percentage.
+        all_freqs = expected_freqs.join(observed_freqs, on='value').dropna(axis=0)
+        all_freqs.loc[:, 'weight'] *= total_count
+
+        # If after dropping there are less then two categories we can't conduct the test.
+        if len(all_freqs) <= 1:
+            logger.warning("Provide weights for at least two categories. Chi-square test is not applicable.")
+            return None
```

Comment:
exception?

Reply:
yes, exceptions probably would be better.